src/MongoDB.Driver.Core/MongoConnectionException.cs

```diff
@@ -78,6 +80,25 @@ public ConnectionId ConnectionId
             get { return _connectionId; }
         }
 
+        /// <summary>
+        /// Whether or not this exception contains a socket timeout exception.
+        /// </summary>
+        public bool ContainsSocketTimeoutException
+        {
+            get
+            {
+                for (var exception = InnerException; exception != null; exception = exception.InnerException)
+                {
+                    if (exception is SocketException socketException &&
+                        socketException.SocketErrorCode == SocketError.TimedOut)
```

Comment:
This could have easily fit on one line... but no need to change it. Just saying.

Reply:
Sorry!